android-json-form-wizard/src/main/java/com/vijay/jsonwizard/activities/JsonFormActivity.java

```diff
@@ -372,39 +365,57 @@ public View getFormDataView(String address) {
     @Override
     public JSONObject getObjectUsingAddress(String[] address, boolean popup) throws JSONException {
         if (address != null && address.length > 1) {
-            if (RuleConstant.RULES_ENGINE.equals(address[0])) {
+            if (RuleConstant.RULES_ENGINE.equals(address[0]) && !address[1].endsWith(".yml")) {
+                JSONArray jsonArray = new JSONArray(address[1]);
+                List<String> keysList = new ArrayList<>();
+                //index 0 has the key
```

Comment:
This assumption may not always be true as above.

You may need to determine if it is indeed a key or not.